apps/asap-server/src/handlers/external-author/index-handler.ts

```diff
@@ -0,0 +1,62 @@
+import { EventBridgeEvent } from 'aws-lambda';
+import { SquidexGraphql } from '@asap-hub/squidex';
+import {
+  AlgoliaSearchClient,
+  algoliaSearchClientFactory,
+} from '@asap-hub/algolia';
+import logger from '../../utils/logger';
+import ExternalAuthors, {
+  ExternalAuthorsController,
+} from '../../controllers/external-authors';
+import {
+  ExternalAuthorEventType,
+  SquidexWebhookExternalAuthorPayload,
+} from '../webhooks/webhook-external-author';
+import { EventBridgeHandler } from '../../utils/types';
+import { algoliaApiKey, algoliaAppId, algoliaIndex } from '../../config';
+
+export const indexExternalAuthorHandler =
+  (
+    externalauthorController: ExternalAuthorsController,
+    algoliaClient: AlgoliaSearchClient,
+  ): EventBridgeHandler<
+    ExternalAuthorEventType,
+    SquidexWebhookExternalAuthorPayload
+  > =>
+  async (event) => {
+    logger.debug(`Event ${event['detail-type']}`);
+
+    try {
+      const externalauthor = await externalauthorController.fetchById(
+        event.detail.payload.id,
+      );
+
+      logger.debug(`Fetched external author ${externalauthor.displayName}`);
+
+      await algoliaClient.saveEntity({
+        id: event.detail.payload.id,
+        type: 'external-author',
+        data: externalauthor,
+      });
+
+      logger.debug(`Saved external author  ${externalauthor.displayName}`);
+    } catch (e) {
+      if (e?.output?.statusCode === 404) {
+        await algoliaClient.remove(event.detail.payload.id);
```

Comment:
Is it guaranteed that ids are unique across users, research outputs and external authors?

Reply:
That is good question.
Maybe @peterstarling @kmaid @markcsims know the answers. I do not

Reply:
Squidex IDs are GUIDs so I guess so?